pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -495,6 +503,18 @@ func (r *ReconcileLogStorage) elasticsearchSecrets(ctx context.Context) ([]*core
 	return secrets, nil
 }
 
+// Returns true if we want to apply a new trial license.
+func (r *ReconcileLogStorage) applyElasticTrialSecret(ctx context.Context) (bool, error) {
```

Comment:
Nit, but this function name sounds like it's actually applying the trial instead of checking if it exists.